src/cursor/abstract_cursor.ts

```diff
@@ -112,11 +112,11 @@ export abstract class AbstractCursor<
   /** @internal */
   [kNamespace]: MongoDBNamespace;
   /** @internal */
-  [kDocuments]: TSchema[];
+  [kDocuments]: WithId<TSchema>[];
```

Comment:
What type issue is requiring the cursor class to be modified? I think nothing here should have to change.
I feel like where ever we create cursors, the `WithId` modifier would pass through the class's parameterization. Using `FindCursor<WithId<TSchema>>` or `FindCursor<TSchema>` would make all the APIs the cursor has return the correct type, I believe?

Reply:
This is used in some functions like on line 235. Reverting the change yields this error:
![image](https://user-images.githubusercontent.com/38259440/141370141-224243b4-af96-448e-9adf-49f530ebebe8.png)